### ISSUE TITLE
Normalized PLL scores

### ISSUE THREAD

**Issue by constanceferragu** [adaptyvbio/competition_metrics#3]
### Issue: 

Currently, the `compute_pll` function calculates the pseudo log-likelihood for a given protein sequence by summing log probabilities across all positions. However, this approach makes the PLL score dependent on sequence length, which complicates comparisons between sequences of different lengths.

### Proposed Solution: 

To address this, I propose modifying `compute_pll` to compute a normalized PLL by dividing the total PLL by the sequence length. This change will provide a length-independent score, representing the average log-likelihood per token in the sequence.

### Suggested Code Change:

Update the `compute_pll` function as follows:

```
def compute_normalized_pll(sequence):
    data = [("protein", sequence)]
    batch_converter = alphabet.get_batch_converter()
    *_, batch_tokens = batch_converter(data)
    log_probs = []
    for i in range(len(sequence)):
        batch_tokens_masked = batch_tokens.clone()
        batch_tokens_masked[0, i + 1] = alphabet.mask_idx
        with torch.no_grad():
            token_probs = torch.log_softmax(
                model(batch_tokens_masked.to(device))["logits"], dim=-1
            )
        log_probs.append(token_probs[0, i + 1, alphabet.get_idx(sequence[i])].item())
    total_log_prob = math.fsum(log_probs)
    normalized_pll = total_log_prob / len(sequence)
    return normalized_pll
```

**Comment by igoradaptyv** [comment]
Okay, I recomputed the ranking based on the per average token PLL vs the sum PLL and checked the correlations 


```
pearson:          rank_cur  rank_new
rank_cur   1.00000   0.99834
rank_new   0.99834   1.00000
kendall:          rank_cur  rank_new
rank_cur  1.000000  0.996629
rank_new  0.996629  1.000000
spearman:          rank_cur  rank_new
rank_cur  1.000000  0.997692
rank_new  0.997692  1.000000
```

These numbers show how similar two different ranking methods are (current vs. new), using three different ways to measure similarity:

1. Pearson (0.998): Measures linear relationship - like "when one goes up, does the other go up too?"
2. Kendall (0.997): Looks at pairs of entries - "are they in the same order in both rankings?"
3. Spearman (0.998): Similar to Pearson but for rankings - "how similar are the overall positions?"

All these numbers are extremely close to 1.0 (perfect match),  so there is barely any difference between using average PLL vs sum PLL

Based on this, we'll keep rankings as is for now because 

- People have already optimized their solutions for the current scoring system
- Changing rules mid-competition wouldn't be fair
- The very high correlation (0.997) means it wouldn't really affect the current standings anyway

We'll def. adopt this for the next round though as you are correct in pointing out that it is the _correct_ way to do this. And thanks for raising the issue :-)
